src/common/wallet-state.ts

```diff
@@ -457,24 +471,47 @@ function useWalletState(initialState?: Partial<WalletStateParams>): WalletData {
   }
 
   const getSyncStatus = async (): Promise<SynchronizationStatus> => {
+    const currentTimestamp = new Date().valueOf()
     if (isMocked) {
       return {
         mode: 'offline',
         currentBlock: 0,
+        lastNewBlockTimestamp: currentTimestamp,
       }
     }
 
+    const previousSyncStatus = syncStatus
     const syncing = await web3.eth.isSyncing()
 
     if (syncing === true) {
       throw Error('Unexpected')
     }
 
+    const currentBlock = await web3.eth.getBlockNumber()
+
+    const lastNewBlockTimestamp =
+      currentBlock === previousSyncStatus.currentBlock
+        ? previousSyncStatus.lastNewBlockTimestamp
+        : currentTimestamp
+
     if (syncing === false) {
-      const currentBlock = await web3.eth.getBlockNumber()
-      return {
-        mode: 'offline',
-        currentBlock: currentBlock,
+      if (
+        currentBlock === 0 ||
```

Comment:
Shouldn't initial block number be fetched when initializing wallet from scratch and that condition changed into `currentBlock === intialBlock`?